corehq/apps/locations/views.py

```diff
@@ -1172,3 +1178,40 @@ def count_locations(request, domain):
     return JsonResponse({
         'count': locations_count
     })
+
+
+@waf_allow('XSS_BODY')
+@csrf_exempt
+@require_POST
+@api_auth()
+def bulk_location_upload_api(request, domain, **kwargs):
+    try:
+        return _bulk_location_upload_api(request, domain)
+    except LocationBulkImportError as e:
+        error, status_code = str(e), 400
+    except Exception as e:
+        notify_exception(None, message=str(e))
+        error, status_code = str(e), 500
+
+    return json_response({'success': False, 'message': error}, status_code=status_code)
+
+
+def _bulk_location_upload_api(request, domain):
+    upload_file = request.FILES.get("bulk_upload_file")
```

Comment:
@szymonbalasz please validate that there is only one file included. Uploading multiple files should be treated as an error.